### PR TITLE
feat: add longest streak to devcard

### PR DESCRIPTION
**Pull request by nensidosari**
- adding longest streak to the devcard, until now we're using only getDevCardData, not getDevCardDataV1
- related PR https://github.com/dailydotdev/apps/pull/3216 

Can't remember if we'd do anything else about image generation with this version

### CODE REVIEW

**Comment by idoshamun** [comment]
Should it be longest or active streak?

**Comment by nensidosari** [comment]
@idoshamun longest says the specs https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/919470082/Reading+streaks+follow+up

**Review by rebelchris — APPROVED** [review]

File: src/common/devcard.ts [hunk 1]
```diff
@@ -82,18 +84,23 @@ export async function getDevCardDataV1(
...
+      select: ['maxStreak'],
+    }),
   ]);
   return {
     user,
     articlesRead,
     tags,
     sourcesLogos: sources.map((source) => source.image),
     rank,
+    longestStreak: streak?.maxStreak ?? 0,
```
Comment (capJavert) [inline src/common/devcard.ts:103]: Can we just make it `maxStreak` as in streak column to avoid confusion in naming?

File: src/common/devcard.ts [hunk 2]
```diff
@@ -128,13 +136,18 @@ export async function getDevCardData(
       })
     ).map(({ value }) => value),
     getFavoriteSources(con, userId),
+    await con.getRepository(UserStreak).findOneOrFail({
```
Comment (capJavert) [inline src/common/devcard.ts:139]: you don't need `await` here

File: __tests__/devcard.ts [hunk 3]
```diff
@@ -331,6 +333,7 @@ describe('query devCard(id)', () => {
       isProfileCover: true,
       showBorder: false,
       articlesRead: 0,
+      longestStreak: 0,
```
Comment (capJavert) [inline __tests__/devcard.ts:336]: Can we also add a test where strak is non 0

File: src/common/devcard.ts (hunk 1 above)
Comment (nensidosari) [reply src/common/devcard.ts:103]: Sure

File: src/common/devcard.ts (hunk 2 above)
Comment (nensidosari) [reply src/common/devcard.ts:139]: removing

**Review by sshanzel — APPROVED** [review]
Don't have anything to add on top of Ante's comments 🥳

File: src/common/devcard.ts (hunk 1 above)
Comment (sshanzel) [reply src/common/devcard.ts:103]: On that note, we can actually avoid making any changes here, we can just add the existing `userStreak` query to the same call.

Comment (capJavert) [reply src/common/devcard.ts]: Yeah, good idea @sshanzel, I agree 👍

**Comment by nensidosari** [comment]
Closing in favour of directly userStreak

**Pull request closed without merging**